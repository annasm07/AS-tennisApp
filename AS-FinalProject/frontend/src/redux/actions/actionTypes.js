export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  SIGN_UP: 'LOG_IN',
  SIGN_UP_ERROR: 'LOG_IN_ERROR',
  GET_PLAYER: 'GET_PLAYER',
  GET_PLAYER_ERROR: 'GET_PLAYER_ERROR',
  GET_PLAYERS: 'GET_PLAYERS',
  GET_PLAYERS_ERROR: 'GET_PLAYERS_ERROR',
  CREATE_MATCH: 'CREATE_MATCH',
  CREATE_MATCH_ERROR: 'CREATE_MATCH_ERROR',
  UPDATE_POINTS: 'UPDATE_POINTS',
  UPDATE_POINTS_ERROR: 'UPDATE_POINTS_ERROR',
  UPDATE_GAMES: 'UPDATE_GAMES',
  UPDATE_GAMES_ERROR: 'UPDATE_GAMES_ERROR',
  END_GAME: 'END_GAME',
  END_GAME_ERROR: 'END_GAME_ERROR',
  END_SET: 'END_SET',
  END_SET_ERROR: 'END_SET_ERROR',
  END_MATCH: 'END_MATCH',
  END_MATCH_ERROR: 'END_MATCH_ERROR',
};
