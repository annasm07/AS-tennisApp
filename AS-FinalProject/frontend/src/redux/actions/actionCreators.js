import axios from 'axios';
import actionTypes from './actionTypes';

const LOG_IN_URL = 'http://localhost:1616/api/auth/login';
const SIGN_UP_URL = 'http://localhost:1616/api/auth/signup';
const GET_PLAYER_URL = 'http://localhost:1616/api/home';
const GET_ALL_PLAYERS = 'http://localhost:1616/api/players';
const CREATE_MATCH_URL = 'http://localhost:1616/api/match';

export function logIn(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}

export function signUp({name, email, password, player, playerName}) {
  return async dispatch => {
    try {
      await axios.post(SIGN_UP_URL, {
        name,
        email,
        password,
        player,
        playerName,
      });
      dispatch({
        // type: actionTypes.SIGN_UP,
        // user: data.user,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.SIGN_UP_ERROR,
      });
    }
  };
}

export function getPlayerInfo(token, playerId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${GET_PLAYER_URL}/${playerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYER,
        player: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYER_ERROR,
      });
    }
  };
}
export function getAllPlayers(token) {
  return async dispatch => {
    try {
      const {data} = await axios(GET_ALL_PLAYERS, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYERS,
        players: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYERS_ERROR,
      });
    }
  };
}

export function newMatch(token, p1Name, p2Name) {
  return async dispatch => {
    try {
      const {data} = await axios.post(
        CREATE_MATCH_URL,
        {p1Name, p2Name},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      );
      dispatch({
        type: actionTypes.CREATE_MATCH,
        currentMatch: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.CREATE_MATCH_ERROR,
      });
    }
  };
}
