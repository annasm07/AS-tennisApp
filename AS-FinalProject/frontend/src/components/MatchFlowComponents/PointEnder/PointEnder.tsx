import React from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {Text, TouchableOpacity, View, StyleSheet} from 'react-native';
import matchButtonsStyles from '../../../theme/matchFlowButtons';
import {counterLogicPoints} from '../../../utils/counterLogic';
import {updatePoints} from '../../../redux/actions/actionCreators';

export default function PointEnder({navigation}: any) {
  const dispatch = useDispatch();
  let currentGamePoints = useSelector((store: any) => store.currentGamePoints);

  function handlePoint(playerWhoWon: String) {
    currentGamePoints = counterLogicPoints(playerWhoWon, currentGamePoints);
    dispatch(updatePoints(currentGamePoints));

    navigation.navigate('ServeButtons');
  }
  return (
    <View style={styles.row}>
      <View
        style={[
          styles.column,
          {
            left: '6%',
          },
        ]}>
        <TouchableOpacity
          style={matchButtonsStyles.button}
          onPress={() => handlePoint('p2')}>
          <Text style={matchButtonsStyles.textRed}>Forced Error</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={matchButtonsStyles.button}
          onPress={() => handlePoint('p2')}>
          <Text style={matchButtonsStyles.textYellow}>Unforced Error</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={matchButtonsStyles.button}
          onPress={() => handlePoint('p1')}>
          <Text style={matchButtonsStyles.textGreen}>Winner</Text>
        </TouchableOpacity>
      </View>
      <View
        style={[
          styles.column,
          {
            left: '18%',
          },
        ]}>
        <TouchableOpacity
          style={matchButtonsStyles.button}
          onPress={() => handlePoint('p1')}>
          <Text style={matchButtonsStyles.textRed}>Forced Error</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={matchButtonsStyles.button}
          onPress={() => handlePoint('p1')}>
          <Text style={matchButtonsStyles.textYellow}>Unforced Error</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={matchButtonsStyles.button}
          onPress={() => handlePoint('p2')}>
          <Text style={matchButtonsStyles.textGreen}>Winner</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  column: {
    marginTop: 10,
  },
  row: {
    flexDirection: 'row',
  },
});
