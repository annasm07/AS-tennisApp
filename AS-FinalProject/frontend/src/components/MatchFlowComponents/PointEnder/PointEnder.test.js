import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../../redux/actions/actionCreators';
import {Provider} from 'react-redux';
import PointEnder from './PointEnder';

jest.mock('../../../redux/actions/actionCreators');

describe('When invoked a PointEnder component - serving player 1', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    server: true,
    secondServe: false,
  });
  let navigation;
  beforeEach(() => {
    jest.spyOn(actions, 'updatePoints').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });
  test('Should render PointEnder component', () => {
    const pointEnder = render(
      <Provider store={store}>
        <PointEnder navigation={navigation} />
      </Provider>,
    );
    expect(pointEnder).toMatchSnapshot();
  });

  test('When UnforecedError1 is pressed, it should navigate to stats', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <PointEnder navigation={navigation} />
      </Provider>,
    );
    const UnforecedError1 = getByTestId('UnforecedError1');
    const handlePoint = jest.fn();
    fireEvent.press(UnforecedError1, handlePoint);
    expect(handlePoint).not.toHaveBeenCalled();
  });

  test('When ForecedError1 is pressed, it should navigate to stats', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <PointEnder navigation={navigation} />
      </Provider>,
    );
    const ForecedError1 = getByTestId('ForecedError1');
    const handlePoint = jest.fn();
    fireEvent.press(ForecedError1, handlePoint);
    expect(handlePoint).not.toHaveBeenCalled();
  });

  test('When Winner1 is pressed, it should navigate to stats', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <PointEnder navigation={navigation} />
      </Provider>,
    );
    const Winner1 = getByTestId('Winner1');
    const handlePoint = jest.fn();
    fireEvent.press(Winner1, handlePoint);
    expect(handlePoint).not.toHaveBeenCalled();
  });

  test('When UnforecedError2 is pressed, it should navigate to stats', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <PointEnder navigation={navigation} />
      </Provider>,
    );
    const UnforecedError2 = getByTestId('UnforecedError2');
    const handlePoint = jest.fn();
    fireEvent.press(UnforecedError2, handlePoint);
    expect(handlePoint).not.toHaveBeenCalled();
  });

  test('When ForecedError2 is pressed, it should navigate to stats', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <PointEnder navigation={navigation} />
      </Provider>,
    );
    const ForecedError2 = getByTestId('ForecedError2');
    const handlePoint = jest.fn();
    fireEvent.press(ForecedError2, handlePoint);
    expect(handlePoint).not.toHaveBeenCalled();
  });

  test('When Winner2 is pressed, it should navigate to stats', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <PointEnder navigation={navigation} />
      </Provider>,
    );
    const Winner2 = getByTestId('Winner2');
    const handlePoint = jest.fn();
    fireEvent.press(Winner2, handlePoint);
    expect(handlePoint).not.toHaveBeenCalled();
  });
});
