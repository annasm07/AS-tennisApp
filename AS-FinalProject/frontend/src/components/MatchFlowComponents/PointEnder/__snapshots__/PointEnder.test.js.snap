// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a PointEnder component - serving player 1 Should render PointEnder component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "left": "6%",
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "display": "flex",
          "height": 140,
          "justifyContent": "center",
          "marginVertical": 5,
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "textAlign": "center",
          "width": 140,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F68A5B",
            "fontSize": 28,
            "textAlign": "center",
          }
        }
        testID="UnforecedError1"
      >
        Unforced Error
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "display": "flex",
          "height": 140,
          "justifyContent": "center",
          "marginVertical": 5,
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "textAlign": "center",
          "width": 140,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F5BF00",
            "fontSize": 28,
            "textAlign": "center",
          }
        }
        testID="ForecedError1"
      >
        Forced Error
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "display": "flex",
          "height": 140,
          "justifyContent": "center",
          "marginVertical": 5,
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "textAlign": "center",
          "width": 140,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8CE576",
            "fontSize": 28,
            "textAlign": "center",
          }
        }
        testID="Winner1"
      >
        Winner
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "left": "18%",
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "display": "flex",
          "height": 140,
          "justifyContent": "center",
          "marginVertical": 5,
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "textAlign": "center",
          "width": 140,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F68A5B",
            "fontSize": 28,
            "textAlign": "center",
          }
        }
        testID="UnforecedError2"
      >
        Unforced Error
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "display": "flex",
          "height": 140,
          "justifyContent": "center",
          "marginVertical": 5,
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "textAlign": "center",
          "width": 140,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F5BF00",
            "fontSize": 28,
            "textAlign": "center",
          }
        }
        testID="ForecedError2"
      >
        Forced Error
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "display": "flex",
          "height": 140,
          "justifyContent": "center",
          "marginVertical": 5,
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "textAlign": "center",
          "width": 140,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8CE576",
            "fontSize": 28,
            "textAlign": "center",
          }
        }
        testID="Winner2"
      >
        Winner
      </Text>
    </View>
  </View>
</View>
`;
